src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -0,0 +1,640 @@
+    cellBlock.setElementType( "C3D" + std::to_string(numPointsPerCell));
+    CellBlock::NodeMapType & cellToVertex = cellBlock.nodeList();
+    cellBlock.resize( numCells );
+    arrayView1d< globalIndex > const & localToGlobal = cellBlock.localToGlobalMap();
+
+    vtkIntArray * attributeDataArray =  getAttributeDataArray();
+
+
+    /// Writing properties
+    for( vtkDataArray * array : arraysTobeImported )
+    {
+      int dimension = array->GetNumberOfComponents();
+      if( dimension == 1 )
+      {
+        real64_array & property = cellBlock.addProperty< real64_array >( array->GetName() );
+        localIndex cellCount = 0;
+        for( vtkIdType c = 0; c < m_vtkMesh->GetNumberOfCells(); c++)
+        {
+          vtkCell * currentCell = m_vtkMesh->GetCell(c);
+          if( currentCell->GetCellType() == cellType )
+          {
+            if( attributeDataArray != nullptr && attributeDataArray->GetValue(c) != region_id) continue;
+            property[cellCount++] = array->GetTuple1(c);
+          }
+        }
+      }
+      else
+      {
+        array2d< real64 > & property = cellBlock.addProperty< array2d< real64 > >( array->GetName() );
+        property.resizeDimension< 1 >( dimension );
+        localIndex cellCount = 0;
+        for( vtkIdType c = 0; c < m_vtkMesh->GetNumberOfCells(); c++)
+        {
+          vtkCell * currentCell = m_vtkMesh->GetCell(c);
+          if( currentCell->GetCellType() == cellType )
+          {
+            if( attributeDataArray != nullptr && attributeDataArray->GetValue(c) != region_id ) continue;
+            for(int i = 0; i < dimension; i++ )
+            {
+              property(cellCount,i) = array->GetComponent(c,i);
+            }
+            cellCount++;
+          }
+        }
+      }
+    }
+
+    /// Writing connectivity and Local to Global
+    cellToVertex.resize( numCells, numPointsPerCell );
+    if( cellType == VTK_HEXAHEDRON ) // Special case for hexahedron because of the ordering
+    {
+      writeHexahedronVertices( cellToVertex, region_id, localToGlobal );
+      return;
+    }
+    if( cellType == VTK_WEDGE ) // Special case for hexahedron because of the ordering
+    {
+      writeWedgeVertices( cellToVertex, region_id, localToGlobal );
+      return;
+    }
+    vtkIdTypeArray * globalCellIdsDataArray = getCellGlobalIdDataArray();
+    localIndex cellCount = 0;
+    for( vtkIdType c = 0; c < m_vtkMesh->GetNumberOfCells(); c++)
+    {
+      vtkCell * currentCell = m_vtkMesh->GetCell(c);
+      if( currentCell->GetCellType() == cellType )
+      {
+        if( attributeDataArray != nullptr && attributeDataArray->GetValue(c) != region_id) continue;
+        for( localIndex v = 0; v < numPointsPerCell; v++)
+        {
+          cellToVertex[cellCount][v] = currentCell->GetPointId(v);
+        }
+        localToGlobal[cellCount] = globalCellIdsDataArray->GetValue(c);
+        cellCount++;
+      }
+    }
```

Comment:
Would be nice to unify all cases and reduce code duplication by having a function that returns a node ordering map for a given VTK cell type. For example, here's what we currently do in VTK output:
https://github.com/GEOSX/GEOSX/blob/496f5b28b5571adfff54d65c727fcf29e1c6ef60/src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp#L75-L90